catroid/src/androidTest/java/org/catrobat/catroid/test/common/LookDataTest.java

```diff
@@ -23,22 +23,90 @@
 import android.support.test.runner.AndroidJUnit4;
+import android.util.Log;
 
 import com.badlogic.gdx.graphics.Pixmap;
 import com.badlogic.gdx.graphics.Texture;
 import com.badlogic.gdx.graphics.g2d.TextureRegion;
 
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.common.Constants;
 import org.catrobat.catroid.common.LookData;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.SingleSprite;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.io.ResourceImporter;
+import org.catrobat.catroid.io.StorageOperations;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.utils.ImageEditing;
+import org.catrobat.catroid.utils.PathBuilder;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import java.io.File;
+import java.io.IOException;
+
+import static junit.framework.Assert.assertEquals;
+
+import static org.catrobat.catroid.common.Constants.IMAGE_DIRECTORY_NAME;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 
 @RunWith(AndroidJUnit4.class)
 public class LookDataTest {
+	private LookData lookData;
+	private String filePath;
+
+	final String expectedMetadata = "0.0;228.0;9.0;321.0;57.0;411.0;136.0;474.0;228.0;500.0;305.0;495.0;375.0;468.0;"
+			+ "436.0;419.0;474.0;364.0;497.0;295.0;499.0;218.0;481.0;151.0;443.0;89.0;385.0;38.0;321.0;9.0;179.0;"
+			+ "9.0;115.0;38.0;57.0;89.0;19.0;151.0|125.0;248.0;154.0;330.0;201.0;365.0;248.0;375.0;313.0;358.0;"
+			+ "365.0;299.0;374.0;234.0;346.0;170.0;285.0;130.0;206.0;133.0;150.0;175.0";
+
+	@Test
+	public void testCollisionInformation() throws IOException, InterruptedException {
+		lookData.getCollisionInformation().loadOrCreateCollisionPolygon();
+
+		String metadata = ImageEditing.readMetaDataStringFromPNG(filePath, Constants.COLLISION_PNG_META_TAG_KEY);
+
+		assertEquals(expectedMetadata, metadata);
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		Project project = new Project(InstrumentationRegistry.getTargetContext(), "testProject");
+		filePath = project.getDefaultScene().getDirectory() + "/" + IMAGE_DIRECTORY_NAME + "/collision_donut.png";
+
+		Sprite sprite = new SingleSprite("testSprite");
+		project.getDefaultScene().addSprite(sprite);
+
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+		XstreamSerializer.getInstance().saveProject(project);
+
+		File imageFile = ResourceImporter.createImageFileFromResourcesInDirectory(
+				InstrumentationRegistry.getContext().getResources(),
+				org.catrobat.catroid.test.R.raw.collision_donut,
+				new File(project.getDefaultScene().getDirectory(), IMAGE_DIRECTORY_NAME),
+				"collision_donut.png",
+				1);
+
+		lookData = new LookData("test", imageFile);
+		sprite.getLookList().add(lookData);
+	}
+
+	@After
+	public void tearDown() {
+		try {
```

Comment:
please don't catch exceptions in tests unless absolutely necessary. it's easier to debug if you just add the exception to the method signature.